pkg/topology/kademlia/kademlia.go

```diff
@@ -281,18 +281,15 @@ func (k *Kad) connectBalanced(wg *sync.WaitGroup, peerConnChan chan<- *peerConnI
 	}
 }
 
-// connectNeighbours attempts to connect to the neighbours
-// which were not considered by the connectBalanced method.
-func (k *Kad) connectNeighbours(wg *sync.WaitGroup, peerConnChan chan<- *peerConnInfo) {
+// connectUntilSaturation attempts to connect to peers until bin oversaturation is reached.
+func (k *Kad) connectUntilSaturation(wg *sync.WaitGroup, peerConnChan chan<- *peerConnInfo) {
```

Comment:
I recommend calling it `connectUntilOversaturation`.

Reply:
I find this change unnecessary. We're mixing things up here. The parameter change should be enough, please confirm @istae

Reply:
If the connector method handles both bins which are inside and outside of depth there can not exist any notion of saturation in the name

Reply:
So how's `connectToPeers` ?

Reply:
this is unfortunately not a naming issue, but an issue of single responsibility. you're not just renaming but mixing up responsibilities. the change you've done here basically overrides the responsibility of the bin balancer by connecting to peers outside of depth, which would mess up the balancing inside the bins.

Reply:
> the neighborhood connecter ignores bins that are out of depth

that is incorrect.

have a look at this conditional that you wrote:
```go
		if bin < k.NeighborhoodDepth() && k.connectedPeers.BinSize(bin) >= saturationPeers {
```
this means that if the bin is out of depth and unsaturated then this connector will indeed cause more peers to be connected outside of depth and this is not what we want. we always want the balancer to do connections out of depth. no reinforcements needed there. if you want more peers out of depth you need to change the number of bits the balancer is trying to brute force.

Reply:
Without a new connector function, the issue of some of the shallower bins never filling up won't be resolved.

Reply:
> the neighborhood connecter ignores bins that are out of depth
yes, `connectNeighbours` in master is doing this,.